### PR TITLE
feat(config): add `snapshotSerializers` option

### LINKED ISSUE #2023: Add customserializer option to vitest.config.ts
### Clear and concise description of the problem

For compatibility with jest, add [customSerializers](https://jestjs.io/docs/configuration#snapshotserializers-arraystring) as a global option. 

## Workaround:

Add a setup function, which runs `expect.addSnapshotSerializer` before every test file.

### Suggested solution

Follow Jest's API

### Alternative

Add a global setup file option, which is just run once during the entire jest process (not for every test file). This file could do something like `expect.addSnapshotSerializer(mySerializer)`.

Con: No Jest compat
Pro: More flexible

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by fenghan34**
### Description

This PR adds the `snapshotSerializers` option to test config and resolves #2023.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: docs/config/index.md [hunk 1]
```diff
@@ -1662,25 +1662,15 @@ Format options for snapshot testing. These options are passed down to [`pretty-f
 ::: tip
 Beware that `plugins` field on this object will be ignored.
 
-If you need to extend snapshot serializer via pretty-format plugins, please, use [`expect.addSnapshotSerializer`](/api/expect#expect-addsnapshotserializer) API.
+If you need to extend snapshot serializer via pretty-format plugins, please, use [`expect.addSnapshotSerializer`](/api/expect#expect-addsnapshotserializer) API or [snapshotSerializers](#snapshotserializers-1-3-0) option.
 :::
 
-### resolveSnapshotPath<NonProjectOption />
```
Comment (hi-ogawa) [inline docs/config/index.md:1668]: It looks like you removed `resolveSnapshotPath` documentation but is it intentional? (like is this a duplicate?)

File: packages/utils/src/diff/index.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import { diffStringsRaw, diffStringsUnified } from './printDiffs'
 import type { DiffOptions } from './types'
 
 export type { DiffOptions, DiffOptionsColor } from './types'
+export type { PrettyFormatPlugin }
```
Comment (hi-ogawa) [inline packages/utils/src/diff/index.ts:24]: Is it necessary to re-export from `@vitest/utils`?
I see you already have same `export type SnapshotSerializer = PrettyFormatPlugin` from `@vitest/snapshot`, so maybe you can use that for internal purpose.

**Review by hi-ogawa — COMMENTED** [review]
Nice implementation and tests! I made a few minor comments.

File: docs/config/index.md (hunk 1 above)
Comment (fenghan34) [reply docs/config/index.md:1668]: Thanks for pointing that out! It was my neglect, already reverted it.

File: packages/utils/src/diff/index.ts (hunk 2 above)
Comment (fenghan34) [reply packages/utils/src/diff/index.ts]: Nice caught! I reused the `SnapshotSerializer` type right now, thanks for your thorough review.

File: packages/vitest/src/runtime/runners/index.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ export async function resolveTestRunner(config: ResolvedConfig, executor: Vitest
     throw new Error('Runner must implement "importFile" method.')
 
   testRunner.config.diffOptions = await loadDiffConfig(config, executor)
+  await loadSnapshotSerializers(config, executor)
```
Comment (sheremet-va) [inline packages/vitest/src/runtime/runners/index.ts:42]: I think we can load it in parallel for better performance:

```ts
const [diffOptions] = await Promise.all([
  loadDiffConfig(config, executor),
  loadSnapshotSerializers(config, executor),
])
```

File: packages/browser/src/client/main.ts [hunk 4]
```diff
@@ -255,6 +257,7 @@ async function runTests(paths: string[], config: ResolvedConfig) {
 
   try {
     runner.config.diffOptions = await loadDiffConfig(config, executor as VitestExecutor)
+    await loadSnapshotSerializers(config, executor as VitestExecutor)
```
Comment (sheremet-va) [inline packages/browser/src/client/main.ts:260]: Load in parallel?

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/runtime/runners/index.ts (hunk 3 above)
Comment (fenghan34) [reply packages/vitest/src/runtime/runners/index.ts]: Make sense!

File: packages/browser/src/client/main.ts (hunk 4 above)
Comment (fenghan34) [reply packages/browser/src/client/main.ts]: Done.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**